api/v1alpha3/common_types.go

```diff
@@ -38,6 +38,10 @@ const (
 	// on the reconciled object.
 	PausedAnnotation = "cluster.x-k8s.io/paused"
 
+	// TemplateClonedFromAnnotation is the machine annotation that stores json marshalled version of the infrastructure template resource
+	// that was cloned for the machine.
+	TemplateClonedFromAnnotation = "cluster.x-k8s.io/cloned-from"
```

Comment:
Given that we want to use string instead of json, should we have `cloned-from-groupkind` and `cloned-from-name` + some utilities to get the information in a struct to compare with later?

Reply:
I think this makes sense, it would help reduce the amount of custom parsing that we would need to do related to string concatenation/splitting with a simple name that can be used as is and a group/kind that we should be able to leverage an upstream parser for. Utilities would help with ensuring that we are consistency when calling from multiple places and also give us some ability to introduce changes with any needed defaulting/fallback needed for backwards compatibility if we decide we need to make changes in the future as well.

Reply:
Suggested `groupkind` there, because it already has a string representation https://github.com/kubernetes/apimachinery/blob/2d992b89ee6927fc5ba834371774ade243cd3732/pkg/runtime/schema/group_version.go#L133 and a utility to parse it https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/schema/group_version.go#L76